corehq/apps/case_search/views.py

```diff
@@ -0,0 +1,31 @@
+import json
+
+from corehq.apps.domain.decorators import require_superuser
+from corehq.apps.domain.views import DomainViewMixin
+from dimagi.utils.web import json_response
+from django.views.generic import TemplateView
+
+
+# Create your views here.
+class CaseSearchView(DomainViewMixin, TemplateView):
+    template_name = 'case_search/case_search.html'
+    urlname = 'case_search'
+
+    @require_superuser
+    def get(self, request, *args, **kwargs):
+        return self.render_to_response(self.get_context_data())
+
+    @require_superuser
+    def post(self, request, *args, **kwargs):
+        from corehq.apps.es.case_search import CaseSearchES
+        query = json.loads(request.POST.get('q'))
+        case_type = query.get('type')
+        search_params = query.get('parameters')
```

Comment:
I was thinking that if `parameters` is missing, this line will succeed, and `search_params` will be None, then line 28 will fail on `for param in search_params:`.
If you want to make `parameters` optional, you'd need to default to `[]` or add in an `if search_params:` statement.  If you DO want a hard failure, this line would be the best place for it in my opinion.